apple/utils.bzl

```diff
@@ -163,9 +163,9 @@ def group_files_by_directory(files, extensions, attr):
                 not_matched = False
                 container = path[:after_index]
                 if container in grouped_files:
-                    grouped_files[container] += [f]
```

Comment:
Could this have alternatively been changed to:

```python
group = grouped_files.get(container, [])
grouped_files[container] = depset(f, transitive = group)
```

I ask this because I'm don't have a strong grasp on when to build up a list, vs when to build up a depset.

Reply:
yeah, that's also equivalent, but since a depset wraps a list, I believe just using a list is cleaner when possible, and then only at the end generate the depset (unless merging the depsets as you go is a cleaner implementation)

Reply:
I think the list approach is cleaner, I'm trying to learn where it's ok to not jump straight to depsets. This gives me a sample point, thanks.